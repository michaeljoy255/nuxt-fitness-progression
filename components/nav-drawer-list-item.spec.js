import { mount, createLocalVue } from '@vue/test-utils'
import Vuetify from 'vuetify'
import NavDrawerListItem from './nav-drawer-list-item'

const localVue = createLocalVue()
let wrapper

beforeEach(() => {
  const vuetify = new Vuetify()
  wrapper = mount(NavDrawerListItem, {
    propsData: {
      route: '/',
      icon: 'person',
      name: 'Hello',
    },
    localVue,
    vuetify,
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Component > nav-drawer-list-item', () => {
  test('matches the snapshot', () => {
    expect(wrapper.html()).toMatchSnapshot()
  })

  test('route prop is a required string', () => {
    expect(NavDrawerListItem.props.route).toMatchObject({
      type: String,
      required: true,
    })
  })

  test('icon prop is a required string', () => {
    expect(NavDrawerListItem.props.icon).toMatchObject({
      type: String,
      required: true,
    })
  })

  test('name prop is a required string', () => {
    expect(NavDrawerListItem.props.name).toMatchObject({
      type: String,
      required: true,
    })
  })

  test('route prop can recieve a value', () => {
    const value = '/example'
    wrapper.setProps({ route: value })
    expect(wrapper.props().route).toBe(value)
  })

  test('icon prop can recieve a value', () => {
    const value = 'group'
    wrapper.setProps({ icon: value })
    expect(wrapper.props().icon).toBe(value)
  })

  test('name prop can recieve a value', () => {
    const value = 'Test Name'
    wrapper.setProps({ name: value })
    expect(wrapper.props().name).toBe(value)
  })
})
