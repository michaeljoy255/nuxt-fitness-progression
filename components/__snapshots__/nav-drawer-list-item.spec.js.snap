// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component > nav-drawer-list-item matches the snapshot 1`] = `
"<a tabindex=\\"0\\" class=\\"v-list-item v-list-item--link theme--light\\">
  <div class=\\"v-list-item__action\\"><i aria-hidden=\\"true\\" class=\\"v-icon notranslate material-icons theme--light\\">person</i></div>
  <div class=\\"v-list-item__content\\">
    Hello
  </div>
</a>"
`;
